source/unix/openssl_platform_init.c

```diff
@@ -493,38 +493,82 @@ static enum aws_libcrypto_version s_resolve_libcrypto_symbols(enum aws_libcrypto
     return found_version;
 }
 
-static enum aws_libcrypto_version s_resolve_libcrypto_lib(void) {
-    const char *libcrypto_102 = "libcrypto.so.1.0.0";
-    const char *libcrypto_111 = "libcrypto.so.1.1";
+static enum aws_libcrypto_version s_libcrypto_version_at_compile_time(void) {
+#ifdef OPENSSL_IS_OPENSSL
+    /*
+     * Currently, this only checks for 1.0.2 vs 1.1. As a future optimization, we can also add a branch for OpenSSL 3.0.
+     * OpenSSL 3.0 is compatible with OpenSSL 1.1, so it works currently.
+     */
+    if (OPENSSL_VERSION_NUMBER < 0x10100000L) {
+        return AWS_LIBCRYPTO_1_0_2;
+    } else {
+        return AWS_LIBCRYPTO_1_1_1;
+    }
+#endif
+    /*
+     * Follow the default path instead of prioritizing the compiled version. This works, since we enforce that
+     * the compiled version and runtime version must be the same for BoringSSL and AWS-LC in
+     * `s_validate_libcrypto_linkage()`.
+     */
+    return AWS_LIBCRYPTO_NONE;
+}
+
+/* Given libcrypto version, return the filename of the .so */
+static char *s_libcrypto_lib_filename(enum aws_libcrypto_version version) {
```

Comment:
trivial: since we "libcrypto_lib" -> "libcrypto_sharedlib" in the other functions...
```suggestion
static char *s_libcrypto_sharedlib_filename(enum aws_libcrypto_version version) {
```

Reply:
Thanks, updated.